src/ringct/rctSigs.cpp

```diff
@@ -317,6 +328,7 @@ namespace rct {
                 toHash[ndsRows + 2 * ii + 2] = L;
             }
             c = hash_to_scalar(toHash);
+            CHECK_AND_ASSERT_MES(!(c == rct::zero()), false, "Bad signature hash");
```

Comment:
It could remove the signer's need to know a private key.

Reply:
But this value is directly from a hash function - the attacker would have to find a pre-image mapping to zero. The check is probably worth it I suppose, because this would break the implementation in the next iteration (nothing is being added on line 326).

Reply:
Yes, it would require a preimage of zero, making it extremely unlikely. One of the Bulletproofs auditors pointed out a similar check was missing in Schnorr signature verification code, and it seemed wise (and cheap) to add it here as well.